addons/website_sale_digital/sale.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details
+
+import logging
+from openerp import models
+
+_logger = logging.getLogger(__name__)
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    def action_invoice_end(self):
+        res = super(SaleOrder, self).action_invoice_end()
+        for order in self:
+            if order.order_line.filtered('product_id.product_tmpl_id.digital_content'):
+                digital_product_template = self.env.ref('website_sale_digital.digital_product_edi_template', raise_if_not_found=False)
```

Comment:
Yes we can. But I've kept it inside loop because we will need send mail only if order contains a digital product. Until that, we don't have to send mail, so no need to check if the template exist or not.